### DISCUSSION TITLE
ESP32 ADC - input voltage range? Is it any Attenuation of ESP32 is used by Tasmota?

### DISCUSSION THREAD

**Discussion by MaximumSU** [arendst/Tasmota#13157] [Q&A] [answered]
ESP32 have the input voltage 0-800mv by default (acording to the docs). Also it have the possibility to extend this range to 3.3v with parameter ADC_ATTEN_DB_11. How it is realized by Tasmota? I can't find any information on this in docs.

**Comment by ascillato** [comment]
Hi, as you didn't have any answer here, you can try addressing this to the [Tasmota Support Chat](https://discord.gg/Ks2Kzd4).

**Comment by pkkrusty** [comment, marked as answer]
Also interested in this topic.  Searching through the code, I don't see that function called anywhere, so my guess is not implemented. ESPHome has implemented [this](https://esphome.io/components/sensor/adc.html), not sure how easy it would be to incorporate that and additional calibration functions into Tasmota.

**Reply by sfromis** [reply]
A quick code search revealed attenuation 11db for ESP32.
https://github.com/arendst/Tasmota/blob/development/tasmota/tasmota_xsns_sensor/xsns_02_analog.ino#L299

**Reply by pkkrusty** [reply]
Ah was looking for the Espressif function in my code search. Thanks.

**Comment by tablatronix** [comment]
So this is hardcoded it seems?
not even a build flag override in ESP32?

I will poke around and see if this can be overriden in framwork, if not it probably needs to be exposed with the prevalence of S2/S3 now

**Reply by sfromis** [reply]
It being hardcoded means that you cannot override it elsewhere. If there is a need, it would seem reasonable to make it available to the usual way of doing Tasmota build time configs.

**Reply by tablatronix** [reply]
Yeah should be simple as adding a define for build flags, I just did a quick glance and do not see anything in esp32 hal that would let you override this externally, it seems it sets all to MAX. Maybe IDF has something
